common/djangoapps/third_party_auth/samlproviderconfig/views.py

```diff
@@ -0,0 +1,71 @@
+"""
+Viewset for auth/saml/v0/samlproviderconfig
+"""
+
+from django.http import Http404
+from django.shortcuts import get_object_or_404
+from edx_rbac.mixins import PermissionRequiredMixin
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from enterprise.models import EnterpriseCustomerIdentityProvider
+from rest_framework import permissions, viewsets
+from rest_framework.authentication import SessionAuthentication
+
+from ..models import SAMLProviderConfig
+from .serializers import SAMLProviderConfigSerializer
+
+
+class SAMLProviderMixin(object):
+    authentication_classes = [JwtAuthentication, SessionAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+    serializer_class = SAMLProviderConfigSerializer
+
+
+class SAMLProviderConfigViewSet(PermissionRequiredMixin, SAMLProviderMixin, viewsets.ModelViewSet):
+    """
+    A View to handle SAMLProviderConfig CRUD
+
+    Usage:
+        [HttpVerb] /auth/saml/v0/provider_config/?enterprise-id=uuid
+
+    permission_required refers to the Django permission name defined
+    in enterprise.rules.
+    The associated rule will allow edx-rbac to check if the EnterpriseCustomer
+    returned by the get_permission_object method here, can be
+    accessed by the user making this request (request.user)
+    Access is only allowed if the user has the system role
+    of 'ENTERPRISE_ADMIN' which is defined in enterprise.constants
+    """
+    permission_required = 'enterprise.can_access_admin_dashboard'
+
+    def get_queryset(self):
+        """
+        Find and return the matching providerconfig for the given enterprise uuid
+        if an association exists in EnterpriseCustomerIdentityProvider model
+        """
+        if self.requested_enterprise_uuid is None:
+            raise Http404('Required enterprise_customer_uuid is missing')
+        enterprise_customer_idp = get_object_or_404(EnterpriseCustomerIdentityProvider,
+            enterprise_customer__uuid=self.requested_enterprise_uuid)
+        return SAMLProviderConfig.objects.filter(pk=enterprise_customer_idp.provider_id)
+
+    @property
+    def requested_enterprise_uuid(self):
+        """
+        The enterprise customer uuid from request params or post body
+        """
+        if self.request.method == "POST":
+            uuid_str = self.request.POST.get('enterprise_customer_uuid')
+            if uuid_str is None:
+              raise Http404('Required enterprise_customer_uuid is missing')
+            return uuid_str
+        else:
+            return self.request.query_params.get('enterprise_customer_uuid')
```

Comment:
Small suggestion: cast the `enterprise_customer_uuid` string to an actual `uuid.UUID4` here, and raise a `ParseError` if it fails: https://www.django-rest-framework.org/api-guide/exceptions/#parseerror
That way, when someone makes a test request with `enterprise_customer_uuid=foo`, it won't raise a server error.

Reply:
thanks, pls check update!!!